session/direct-ipc.c

```diff
@@ -0,0 +1,241 @@
+#define _POSIX_C_SOURCE 200809L
+#include <errno.h>
+#include <stdio.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include <sys/socket.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <sys/sysmacros.h>
+#include <sys/wait.h>
+#include <xf86drm.h>
+#include <linux/major.h>
+#include <wlr/util/log.h>
+#include "session/direct-ipc.h"
+
+enum { DRM_MAJOR = 226 };
+
+#ifdef HAS_LIBCAP
+#include <sys/capability.h>
+
+static bool have_permissions(void) {
+	cap_t cap = cap_get_proc();
+	cap_flag_value_t val;
+
+	if (!cap || cap_get_flag(cap, CAP_SYS_ADMIN, CAP_PERMITTED, &val) || val != CAP_SET) {
```

Comment:
The last line that appears the in the log is https://github.com/SirCmpwn/wlroots/blob/master/examples/shared.c#L480 with socket printing as (null), so I think it's completely unrelated to this. It probably has something to do with environment variables or something that wl_display_add_socket_auto expects, but I'm not really sure.

It's up to you if you want that to be part of that to be part of this pull request or a separate fix.